pkg/vcs/git.go

```diff
@@ -635,3 +635,26 @@ func (git *git) CommitExists(commit string) (bool, error) {
 	}
 	return true, nil
 }
+
+func (git *git) PushCurrentCommitAsTagTo(repo string, gitConfig ...string) error {
+	if repo == "" {
```

Comment:
Nit: would it be better to explicitly check for `mgr.cfg.GitCache` before calling the method?

Reply:
Upper level function looks complex. I wanted to remove some checks from there if possible.
Like we do in uploadCoverReport + uploadCoverStat + uploadCorpus.

Reply:
> Upper level function looks complex

It is, unfortunately. But we can always create some helper function in `job.go` that would do the check and archive the commit, if necessary.

Reply:
It was my first version. 3 lines long check here looks better than 1 more function there.
ArchiveCommit is just 21 lines long now.
Do you insist?

Reply:
I marked it as a `Nit:`, so it's a non-critical comment anyway :)